#699 Side panel element alignment in common-properties

nmgokhale opened the pull request:
Fixes https://github.com/elyra-ai/canvas/issues/699
All changes in this PR are listed in this comment - https://github.com/elyra-ai/canvas/issues/699#issuecomment-872617946
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

matthoward366 reviewed (changes requested):
![Screen Shot 2021-07-08 at 9 09 47 AM](https://user-images.githubusercontent.com/21126000/124936520-417d8680-dfcc-11eb-8772-723e836f37de.png)

For some apps this is what these changes will do.

nmgokhale commented:
If we keep left margin on twisty label, content and label is not aligned - 

![Screen Shot 2021-07-08 at 3 00 37 PM](https://user-images.githubusercontent.com/25124000/124996191-64745e80-dffd-11eb-95a1-bf202d45b841.png)

matthoward366 commented on canvas_modules/common-canvas/src/common-properties/panels/control/control.scss:
```diff
@@ -25,6 +28,14 @@
+	}
+}
+
+.properties-control-panel .properties-control-panel .properties-control-panel {
```
What is this styling used for?  A comment might be good here.

nmgokhale replied:
Removed this styling. It was not used anywhere.

nmgokhale commented:
After overwriting carbon CSS for accordion, twisty panel looks like this -
![Screen Shot 2021-07-13 at 3 29 24 PM](https://user-images.githubusercontent.com/25124000/125533423-7ca49baa-17c6-433d-8f74-307312384b6b.png)


![Screen Shot 2021-07-13 at 3 29 32 PM](https://user-images.githubusercontent.com/25124000/125533425-63b53726-97f1-4130-bf1b-667129613d2f.png)

matthoward366 commented on canvas_modules/common-canvas/src/common-properties/panels/action-panel/action-panel.jsx:
```diff
@@ -20,12 +20,19 @@ import { connect } from "react-redux";
+				className={classNames(
+					"properties-action-panel",
+					{ "hide": this.props.panelState === STATES.HIDDEN },
+					{ "properties-control-nested-panel": get(this.props.panel, "nestedPanel", false) },
```
I think you can just use `this.props.panel.nestedPanel` since you already default this in the form generation.

matthoward366 commented on canvas_modules/common-canvas/src/common-properties/panels/column/column.jsx:
```diff
@@ -42,7 +43,13 @@ class ColumnPanel extends React.Component {
+				className={classNames(
+					"properties-column-panel",
+					{ "hide": this.props.panelState === STATES.HIDDEN },
+					{ "properties-control-nested-panel": get(this.props.panel, "nestedPanel", false) },
```
Same as above

matthoward366 commented on canvas_modules/common-canvas/src/common-properties/panels/control/control.jsx:
```diff
@@ -20,14 +20,21 @@ import { connect } from "react-redux";
+				className={classNames(
+					"properties-control-panel",
+					{ "hide": this.props.panelState === STATES.HIDDEN },
+					{ "properties-control-nested-panel": get(this.props.panel, "nestedPanel", false) },
```
Same as above

matthoward366 commented on canvas_modules/common-canvas/src/common-properties/panels/summary/summary.jsx:
```diff
@@ -305,7 +305,13 @@ class SummaryPanel extends React.Component {
+			"properties-summary-panel",
+			"properties-control-panel",
+			{ "hide": this.props.panelState === STATES.HIDDEN },
+			{ "properties-control-nested-panel": get(this.props.panel, "nestedPanel", false) },
```
same as above

matthoward366 commented on canvas_modules/common-canvas/src/common-properties/panels/text-panel/text-panel.jsx:
```diff
@@ -29,7 +30,13 @@ class TextPanel extends Component {
+				className={classNames(
+					"properties-text-panel",
+					{ "hide": this.props.panelState === STATES.HIDDEN },
+					{ "properties-control-nested-panel": get(this.props.panel, "nestedPanel", false) },
```
same as above

matthoward366 commented on canvas_modules/common-canvas/src/common-properties/panels/twisty/twisty.jsx:
```diff
@@ -21,14 +21,21 @@ import classNames from "classnames";
+				className={classNames(
+					"properties-twisty-panel",
+					{ "hide": this.props.panelState === STATES.HIDDEN },
+					{ "properties-control-nested-panel": get(this.props.panel, "nestedPanel", false) },
```
Same as above

matthoward366 commented:
textPanels don't look right.  Is the label and description suppose to have that much space.
![Screen Shot 2021-07-22 at 3 36 37 PM](https://user-images.githubusercontent.com/21126000/126706222-6e45920e-559a-4058-a41d-7c63fde9bc0c.png)

In the panelCondition_paramDef.json can you add the subpanel attribute so we can see which panels are subpanels.

nmgokhale commented:
Addressed all review comments. 

For text panels, spacing between label & description is now updated to `$spacing-03`(8px).
In previous commits, it was `$spacing-05` (16px). 

![Screen Shot 2021-07-22 at 4 39 34 PM](https://user-images.githubusercontent.com/25124000/126721942-692416c1-d43e-4267-b993-41fae35089c3.png)

matthoward366 commented on canvas_modules/common-canvas/src/common-properties/panels/text-panel/text-panel.scss:
```diff
@@ -15,7 +15,9 @@
 
 .properties-text-panel {
-	padding: $spacing-04 0 $spacing-04 0;
+	:not(:empty) {
```
When with this take effect?  I'm wondering if we should have this or not.

nmgokhale replied:
This condition doesn't add spacing when text-panel is empty. 
For example, in `panelConditions_paramDef.json` -

:not(:empty) condition added -
<img width="550" alt="Screen Shot 2021-07-29 at 1 53 59 PM" src="https://user-images.githubusercontent.com/25124000/127565065-fd706de8-98fb-47b5-acc9-a8754d062d9c.png">

:not(:empty) condition removed -
<img width="616" alt="Screen Shot 2021-07-29 at 1 57 10 PM" src="https://user-images.githubusercontent.com/25124000/127565067-0b9f5d67-ad8e-4eb5-aa01-503ef9f9db0b.png">

matthoward366 reviewed (commented):
One comment.  The rest looks good.  We'll need to update the uihints schema with the new attribute.

matthoward366 reviewed (approved):
Looks good.
